### PR TITLE
feat: get top spoken languages

### PR DESCRIPTION
**Pull request by GeronimoJohn**
# Description

Depends on #3453 

### Issue

We need the ability to get the top spoken languages available to a `Country` and the content that we have in `Algolia`

### Solution

Create a function to get the top spoken languages of a country based on the number of speakers. The logic is based on the number of speakers who speak a particular language. We then retrieve these languages and check if they exist in the `languageEnglishName` refinements.

### CODE REVIEW

File: libs/journeys/ui/src/libs/algolia/getTopSpokenLanguages/getTopSpokenLanguages.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  const languageNames = sortedLanguages.map(({ language }) => {
+    const localName = language.name.find(({ primary }) => !primary)?.value
+    const nativeName = language.name.find(({ primary }) => primary)?.value
+    return localName ?? nativeName ?? ''
+  })
+
+  const topSpokenLanguages = languageNames
+    .filter(Boolean)
+    .filter((language) => availableLanguageSet.has(language))
+    .slice(0, 4)
```
Comment (scottkelso) [inline libs/journeys/ui/src/libs/algolia/getTopSpokenLanguages/getTopSpokenLanguages.tsx]: Could you add a function parameter that defaults to 4 instead, therefore allowing anyone to override this if anyone needs to.

File: libs/journeys/ui/src/libs/useCountryQuery/useCountryQuery.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+          }
+        }
+        speakers
+      }
+    }
+  }
+`
+
+export function useCountryQuery(
+  variables?: GetCountryVariables
```
Comment (scottkelso) [inline libs/journeys/ui/src/libs/useCountryQuery/useCountryQuery.ts:24]: Should variables be optional here? `const { data } = useCountryQuery()` yields empty response. Isn't this query only useful when we provide an ID like `const { data } = useCountryQuery({ countryId: 'AD' })`?

File: libs/journeys/ui/src/libs/useCountryQuery/useCountryQuery.mock.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import { GetCountry } from './__generated__/GetCountry'
+import { country } from './data'
+import { GET_COUNTRY } from './useCountryQuery'
+
+export const getCountryMock: MockedResponse<GetCountry> = {
+  request: {
+    query: GET_COUNTRY,
+    variables: {
+      countryId: 'country.id'
```
Comment (scottkelso) [inline libs/journeys/ui/src/libs/useCountryQuery/useCountryQuery.mock.ts:11]: Just a thought - Would it be more useful to use something that is at least the same shape as the real data? E.g. `XX`?

**Review by scottkelso — CHANGES_REQUESTED** [review]
Nice! Just left some suggestions.

File: libs/journeys/ui/src/libs/algolia/getTopSpokenLanguages/getTopSpokenLanguages.tsx (hunk 1 above)
Comment (GeronimoJohn) [reply libs/journeys/ui/src/libs/algolia/getTopSpokenLanguages/getTopSpokenLanguages.tsx]: Love it, this would be following Open/Closed principle right?

**Review by scottkelso — APPROVED** [review]
Awesome!

**Pull request merged**